instrumentation/message-handler/message-handler-1.0/library/src/main/java/io/opentelemetry/instrumentation/messagehandler/MessageHandler.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.messagehandler;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import java.util.Collection;
+
+public abstract class MessageHandler<INPUT> {
+
+  protected abstract Instrumenter<Collection<INPUT>, Void> getMessageInstrumenter();
+
+  protected abstract void doHandle(Collection<INPUT> request);
+
+  public void handle(Collection<INPUT> request) {
```

Comment:
This seems to do nothing messaging specific. I wonder if there is already an utility method like `callInstrumented(input, instrumenter, action)` that could be used.

Or maybe this should iterate over the collection?

Reply:
I am trying to pull out as much abstraction as I can between the two SQSMessageHandlers.

This is for handling batches of messages. We iterate over the messages to add span links, but for handling the event we should only call the doHandle method once.

I looked in the code base for "Instrumenter Runnable/Future", but I do not see a generic utility method.